openshift/tests-extension/test/webhooks.go

```diff
@@ -0,0 +1,362 @@
+			}).WithTimeout(1 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+
+			By("checking webhook is responsive through secret recreation after manual deletion")
+			tlsSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      certificateSecretName,
+					Namespace: webhookOperatorInstallNamespace,
+				},
+			}
+			err := k8sClient.Delete(ctx, tlsSecret, client.PropagationPolicy(metav1.DeletePropagationBackground))
+			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+
+			By("waiting for the webhook operator's service certificate secret to be recreated and populated")
+			Eventually(func(g Gomega) {
+				secret := &corev1.Secret{}
+				err := k8sClient.Get(ctx, client.ObjectKey{Name: certificateSecretName, Namespace: webhookOperatorInstallNamespace}, secret)
+				if apierrors.IsNotFound(err) {
+					GinkgoLogr.Info(fmt.Sprintf("Secret %s/%s not found yet (still polling for recreation)", webhookOperatorInstallNamespace, certificateSecretName))
+					return
+				}
+				g.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to get webhook service certificate secret %s/%s: %v", webhookOperatorInstallNamespace, certificateSecretName, err))
+				g.Expect(secret.Data).ToNot(BeEmpty(), "expected webhook service certificate secret data to not be empty after recreation")
+			}).WithTimeout(2*time.Minute).WithPolling(10*time.Second).Should(Succeed(), "webhook service certificate secret did not get recreated and populated within timeout")
+
+			Eventually(func(g Gomega) {
+				resourceName := fmt.Sprintf("tls-deletion-test-%s", rand.String(5))
+				resource := newWebhookTestV1(resourceName, webhookOperatorInstallNamespace, true)
+
+				_, err := dynamicClient.Resource(webhookTestGVRV1).Namespace(webhookOperatorInstallNamespace).Create(ctx, resource, metav1.CreateOptions{})
+				g.Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to create test resource %s: %v", resourceName, err))
+
+				err = dynamicClient.Resource(webhookTestGVRV1).Namespace(webhookOperatorInstallNamespace).Delete(ctx, resource.GetName(), metav1.DeleteOptions{})
+				g.Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred(), fmt.Sprintf("failed to delete test resource %s: %v", resourceName, err))
+			}).WithTimeout(2 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
+		})
+	})
+
+var webhookTestGVRV1 = schema.GroupVersionResource{
+	Group:    "webhook.operators.coreos.io",
+	Version:  "v1",
+	Resource: "webhooktests",
+}
+
+var webhookTestGVRV2 = schema.GroupVersionResource{
+	Group:    "webhook.operators.coreos.io",
+	Version:  "v2",
+	Resource: "webhooktests",
+}
+
+func newWebhookTestV1(name, namespace string, valid bool) *unstructured.Unstructured {
+	mutateValue := valid
+	obj := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "webhook.operators.coreos.io/v1",
+			"kind":       "WebhookTest",
```

Comment:
```suggestion
			"kind":       "TestWebhook",
```

Reply:
Matter of preference: I am oK with but we can do in a follow up

Reply:
We cannot do that. It will fail does not match with the orginal as well
Please, see the original PR. The kind must match with what is in the sample.